sixMans/game.py

```diff
@@ -192,9 +257,213 @@ async def process_captains_pick(self, reaction, user):
+        player_scores = self.get_player_scores()
+        team_combos = self.get_team_combos()
+
+        # Calc perfectly balanced team based on scores
+        score_total = 0
+        for player, p_data in player_scores.items():
+            score_total += p_data['Score']
+        avg_team_score = score_total/(len(self.players)/2)
+
+        # Determine balanced teams
+        balanced_teams = []
+        balance_diff = None
+        for a_team in team_combos:
+            team_score = 0
+            for player in a_team:
+                team_score += player_scores[player]['Score']
+            
+            team_diff = abs(avg_team_score - team_score)
+            if balance_diff:
+                if team_diff < balance_diff:
+                    balance_diff == team_diff
+                    balanced_teams = [a_team]
+                elif team_diff == balance_diff:
+                    balanced_teams.append(a_team)
+            else:
+                balance_diff == team_diff
+                balanced_teams = [a_team]
+        
+        # return balanced team
+        return balanced_teams, team_diff
+
+    def get_player_scores(self):
+        # Get Player Stats
+        scores = {}
+        ranked_players = 0
+        rank_total = 0
+        wp_players = 0
+        wp_total = 0
+        for player in self.players:
+            player_stats = self.queue.get_player_summary(player)
+
+            rank = 1  # get_player_rank
+            if player_stats:
+                p_wins = player_stats['Wins']
+                p_losses = player_stats['GamesPlayed'] - p_wins
+                qwp = round(self._get_wp(p_wins, p_losses), 2)
+            else:
+                qwp = None
+
+            scores[player] = {"Rank": rank, "QWP": qwp}
+            if rank: 
+                ranked_players += 1
+                rank_total += rank 
+            if qwp:
+                wp_players += 1
+                wp_total += qwp
+        
+        rank_avg = rank_total/ranked_players if ranked_players else 1
+        
+        # Score Players, Avg
+        score_total = 0
+        for player, p_data in scores.items():
+            p_rank = p_data['Rank'] if ('Rank' in p_data and p_data['Rank']) else rank_avg
+            p_wp = p_data['QWP'] if ('QWP' in p_data and p_data['QWP']) else 0.5
+            score_adj = (p_wp * 2) - 1  # +/- 1
+            
+            score = p_rank + score_adj 
+            p_data['Score'] = score
+            score_total += score
+        return scores 
+
+    def get_team_combos(self):
+        team_lineups = []
+        players = self.players
+        for a_player in players:
+            for b_player in players:
+                for c_player in players:
+                    lineup = list(set([a_player, b_player, c_player]))
+                    # lineup.sort()
+                    lineup = sorted(lineup, key=operator.attrgetter('id'))
+                    if (len(lineup) == len(players)/2) and lineup not in team_lineups:
```

Comment:
Does this handle both 2 player teams and 3 correctly? I'm not 100% on how this works but it seems you're creating an initial lineup with 3 players and then checking if that's half the players and doing nothing if it's not.

Reply:
As of now, no, it only handles 3v3. I plan to improve upon this for the next iteration with an **nCr** function

Reply:
Added changes to address this now :)